### ISSUE TITLE
Spurious NU5128 warning related to content in runtimes folder

### ISSUE THREAD

**Issue by rseanhall** [NuGet/Home#10665]
## Details about Problem

NuGet product used (NuGet.exe | Visual Studio | MSBuild.exe | dotnet.exe):

> Microsoft (R) Build Engine version 16.9.0+57a23d249 for .NET Framework
> .NET SDK 5.0.201 a09bd5c86c

## Repro steps and/or sample project

1. Clone https://github.com/rseanhall/Core.Native/tree/nu5128-bug
2. Run appveyor.cmd

Observe no warnings or errors.

3. Reset to the previous commit (https://github.com/rseanhall/Core.Native/commit/4e3c12a45cb91828ac94963129495033a4402136).
4. Run appveyor.cmd

Observe the NU5128 warning.

5. Reset back 5 more commits (https://github.com/rseanhall/Core.Native/commit/8296f237c69743d3e6cbdc58fc07e8470562b92e).
6. Run appveyor.cmd

Observe no warnings or errors.

### Summary

We are packing a .nupkg through an SDK-style .csproj. This has native assets that require us to author the .nuspec ourselves. The native assets are properly put into the runtimes folder. If those files are moved under the lib/netstandard2.0 folder, then there is no warnings but the package is broken because they belong in the runtimes folder.

### Verbose Logs

```
Build succeeded.

"C:\src\mynewwix4\Core.Native\src\WixToolset.Core.Native\WixToolset.Core.Native.csproj" (Pack target) (1) ->
(GenerateNuspec target) ->
  C:\Program Files\dotnet\sdk\5.0.201\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): warning
NU5128: Some target frameworks declared in the dependencies group of the nuspec and the lib/ref folder do not have exac
t matches in the other location. Consult the list of actions below: [C:\src\mynewwix4\Core.Native\src\WixToolset.Core.N
ative\WixToolset.Core.Native.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): warning NU
5128: - Add lib or ref assemblies for the netstandard2.0 target framework [C:\src\mynewwix4\Core.Native\src\WixToolset.
Core.Native\WixToolset.Core.Native.csproj]

    1 Warning(s)
    0 Error(s)
```

**Comment by zkat** [comment]
Native assets should not go under the `lib/` folder. They should instead go either under `runtimes/` or `build/native` with a `.targets` file.

**Comment by rseanhall** [comment]
@zkat Of course. The only reason I put it under the `lib/` folder was to show the minimal change to make the warning go away - implying that the warning is incorrectly complaining about something in the `runtimes/` folder.

**Comment by zkat** [comment]
@heng-liu can you take a look at this?

**Comment by zkat** [comment]
@dominoFire can you take a look at this?

**Comment by nkolev92** [comment]
The warning is actually getting raised for the `snupkg` rather than the nupkg. 

I don't think this particular warning should be run for snupkgs.
